client/src/app/components/Forms/Login/index.tsx

```diff
@@ -0,0 +1,124 @@
+'use client'
+import React, { useState } from 'react'
+import { FaRegEye, FaRegEyeSlash } from 'react-icons/fa'
+import { useForm } from 'react-hook-form';
+import { yupResolver } from '@hookform/resolvers/yup';
+import { loginUserValidation } from '@/app/register/validation';
+import { RESULT_STATUS } from '@/constants';
+import { useRouter } from 'next/navigation';
+import Link from 'next/link';
+import { NextResponse } from 'next/server';
+
+const LoginFormComponent = () => {
+    const router = useRouter()
+    const [showPassword, setShowPassword] = useState(false);
+    const [errorMessage, setErrorMessage] = useState("");
+
+    const togglePasswordVisibility = () => {
+        setShowPassword(!showPassword);
+    };
+    const { register, handleSubmit, formState: { errors } } = useForm({ resolver: yupResolver(loginUserValidation) })
+    const onSubmit = async (data: any) => {
+        try {
+            await loginUserValidation.isValidSync({ ...data }, { abortEarly: false });
+
+            const response = await fetch('api/login', {
+                method: 'POST',
+                headers: {
+                    'Content-Type': 'application/json'
+                },
+                body: JSON.stringify(data)
+            });
+
+            const responseData = await response.json();
+
+
+            if (responseData.status === RESULT_STATUS.SUCCESS) {
+
+                const token = responseData.data.token;
+                document.cookie = `token=${token}; path=/`;
+                router.push('/task')
+
+                return NextResponse.json(data);
+            }
+            if (responseData.status === RESULT_STATUS.FAILURE) {
+                setErrorMessage(responseData.message);
+            }
```

Comment:
check this logic again. you are already setting cookie from route handler and here you are again setting cookie